recipe-server/client/control_new/components/recipes/RecipeListing.js

```diff
@@ -156,9 +158,27 @@ export default class RecipeListing extends React.Component {
     push(getCurrentURL({ page }));
   }
 
-  handleRowClick(record) {
-    const { push } = this.props;
-    push(`/recipe/${record.id}`);
+  handleRowClick(record, index, event) {
+    const linkTarget = event.target.tagName === 'A';
+
+    // If the user has clicked a link directly, just fall back to the native event.
+    if (linkTarget) {
+      return;
+    }
+
+    // If we're here, the user clicked the row itself, which now needs to behave
+    // as if it was a native link click. This includes opening a new tab if using
+    // a modifier key (like ctrl).
+
+    const usingModifierKey = (event.ctrlKey || event.metaKey || event.button === 1);
```

Comment:
This isn't accurately named. It doesn't respect all modifier keys (like shift) and middle clicking isn't a modifier. Maybe `shouldOpenNewTab`? Alternatively, it would make sense to fold this logic into the if statement below, if `!linkTarget` goes away.